perf(linter/plugins): reduce var assignments

overlookmotel opened the pull request:
Follow-on after #15861. Reduce variable assignments by using the final value var as `hi` in the binary search.

Also, because we initialize `sliceEnd` before the loop, don't need to check if it's `undefined` later on.

lilnasy reviewed (commented):
LGTM. Can't wait for these performance improvements to look tiny against our full implementation.

overlookmotel committed:
perf(linter/plugins): reduce var assignments (#15953)

Follow-on after #15861. Reduce variable assignments by using the final value var as `hi` in the binary search.

Also, because we initialize `sliceEnd` before the loop, don't need to check if it's `undefined` later on.
